File: exercism_data/src/684.rb
class Phrase
  attr_reader :phrase

  def initialize(phrase)
    @phrase = phrase
  end

  def word_count
    words = phrase.downcase.split(/[^\w']/) # regex matches any non alphanumeric characters except apostrophes

    words.each_with_object({}) do |word, hsh|
      hsh[word] = words.count { |w| word.eql? w } unless word.empty?
    end
  end

end

File: assignments/python/wc/src/684.py
def word_count(phrase):
	punc = ['!',':',',','@','#','$','%','^','&','*']
	for char in punc:
		phrase = phrase.replace(char, '')
	wordcounts = {}
	words = phrase.split()
	for word in words:
		word = word.lower()
		if word not in wordcounts:
			wordcounts[word] = 0
		wordcounts[word]+=1
	return wordcounts
